wrapanapi/systems/virtualcenter.py

```diff
@@ -162,6 +162,41 @@ def host(self):
         self.refresh()
         return self.raw.runtime.host.name
 
+    @staticmethod
+    def _progress_log_callback(logger, source, destination, progress):
+        logger.info("Provisioning progress {}->{}: {}".format(
+            source, destination, str(progress)))
+
+    @staticmethod
+    def _get_loc_of_vm(source_template, progress_callback):
+        """ Get the location where the inventory object will be stored"""
+        try:
+            folder = source_template.parent.parent.vmParent
+        except AttributeError:
+            folder = source_template.parent
+        progress_callback("Picked folder `{}`".format(folder.name))
+
+        return folder
+
+    @staticmethod
+    def _set_vm_clone_spec(power_on, template, cpu, ram, vm_reloc_spec):
+        """Set properties for Virtual Machine Cloning Operation specification """
+
+        vm_clone_spec = vim.VirtualMachineCloneSpec()
+
+        vm_clone_spec.powerOn = power_on
+        vm_clone_spec.template = template
+        vm_clone_spec.location = vm_reloc_spec
+        vm_clone_spec.snapshot = None
+
+        # Optional arguments
+        if cpu is not None:
```

Comment:
Q: if these arguments are optional, why aren't they set to defaults of None in the func definition? `cpu=None, ram=None`

Reply:
They are set to none here: https://github.com/ManageIQ/wrapanapi/pull/368/files#diff-d3e14fb23e4e2780e08340299eda2cd1R410